src/sundials/sundials_memory.c

```diff
@@ -203,23 +203,44 @@ int SUNMemoryHelper_CopyAsync(SUNMemoryHelper helper, SUNMemory dst,
 
 int SUNMemoryHelper_Destroy(SUNMemoryHelper helper)
 {
-  if (helper->ops->destroy == NULL)
+  if (!helper) return 0;
+
+  if (helper->ops)
```

Comment:
Can `helper->ops` really ever be `NULL` and it not be an error (i.e., outside of this function)? If the answer is no, then I think you can change line 206 to
```c
if (!helper || !helper->ops) return 0;
```
and simplify the logic below.

Reply:
This shouldn't happen unless there is an error when constructing the helper and this function is then called to destroy up the partially constructed object. So I'm inclined to leave the extra check if only to cover this corner case.

Reply:
We should chat on Tuesday about the error handling updates. Specifically, I think we should remove checks like these and almost uniformly trust that the "class" objects are not corrupt/null. The short version of my reasoning is that it is not an error that will be recoverable from in any case I can think of in SUNDIALS, so it is better to let a segfault happen (which is very easy to debug, while, for example, it is much harder to debug why a function returned NULL). I can elaborate at our meeting.

Reply:
I was thinking... for `Destroy` functions we should definitely check if the object is `NULL` or not so we can emulate `delete` rather than `free` (i.e., dont error out if the object is `NULL`). However, I think the check to make sure the ops is not `NULL` is not needed.